Mage/src/main/java/mage/target/targetadjustment/EachOpponentPermanentTargetsAdjuster.java

```diff
@@ -0,0 +1,51 @@
+package mage.target.targetadjustment;
+
+import mage.abilities.Ability;
+import mage.filter.Filter;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+import mage.util.Copyable;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public class EachOpponentPermanentTargetsAdjuster implements TargetAdjuster, Copyable<TargetAdjuster> {
+    TargetPermanent baseTarget;
+
+    public EachOpponentPermanentTargetsAdjuster(TargetPermanent baseTarget) {
+        this.baseTarget = baseTarget;
+    }
+
+    protected EachOpponentPermanentTargetsAdjuster(final EachOpponentPermanentTargetsAdjuster targetAdjuster) {
+        this.baseTarget = targetAdjuster.baseTarget.copy();
+    }
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        ability.getTargets().clear();
+        for (UUID opponentId : game.getOpponents(ability.getControllerId())) {
+            Player opponent = game.getPlayer(opponentId);
+            if (opponent == null) {
+                continue;
+            }
+            TargetPermanent newTarget = baseTarget.copy();
+            Filter<Permanent> filter = newTarget.getFilter();
+            filter.add(new ControllerIdPredicate(opponentId));
+            if (newTarget.canChoose(ability.getControllerId(), ability, game)) {
+                filter.setMessage(filter.getMessage()+" controlled by " + opponent.getLogName());
+                ability.addTarget(newTarget);
+            }
+        }
+    }
+
+    @Override
+    public EachOpponentPermanentTargetsAdjuster copy() {
```

Comment:
is this needed? all other target adjusters are enum. I don't think `copy()` is ever being called?

Reply:
Previous target adjustors never stored any info since the target definition was hard-coded, letting them be an enum where this must be a class. I'm not sure if it's *needed* but it seemed like the safer option to have it.

Reply:
but what is it overriding?? It's a functional interface extending Serializable.

Reply:
The function is from `Copyable<TargetAdjuster>`, but yeah I guess that isn't needed since watchers shouldn't be storing them directly.